src/main/java/com/boomi/flow/services/boomi/mdh/common/Entities.java

```diff
@@ -55,21 +55,14 @@ public static void AddRandomUniqueId(MObject object, String idField) {
 
         // Set the object's external ID too, which is only used inside Flow itself
         object.setExternalId(id);
+
+        return object;
     }
 
     private static List<Property> createPropertiesModel(String id, Map<String, Object> map) {
         return map.entrySet().stream()
                 // Todo: we ignored child objects until get support in engine for child object bindings
                 .filter(field -> field.getValue() instanceof Map == false)
-                .map(field -> {
-                    if (field.getValue() instanceof Map == false) {
-
-                        return new Property(field.getKey(), field.getValue());
-                    } else {
-                        MObject object = new MObject(field.getKey(), id, createPropertiesModel(id, (Map<String, Object>) field.getValue()));
-
-                        return new Property(field.getKey(), object);
-                    }
-                }).collect(Collectors.toList());
+                .map(field -> new Property(field.getKey(), field.getValue())).collect(Collectors.toList());
```

Comment:
Can you move this `collect` onto a new line so it's clear that it's being called not as part of the `map`?